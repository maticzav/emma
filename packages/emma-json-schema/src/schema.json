{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "JSON schema for emma.json files",
  "properties": {
    "boilerplates": {
      "type": "array",
      "items": {
        "type": "string"
      }
    }
  },
  "required": ["boilerplates"],
  "additionalProperties": false
}
