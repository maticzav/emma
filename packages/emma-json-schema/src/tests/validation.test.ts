import * as Ajv from 'ajv'
import schema = require('../schema.json')
import * as fs from 'fs'
import * as path from 'path'

// SETUP

const ajv = new Ajv().addMetaSchema(
  require('ajv/lib/refs/json-schema-draft-06.json'),
)
const validateSchema = ajv.compile(schema)

export function parseConfig(path: string) {
  const file = fs.readFileSync(path, 'utf-8')

  if (!validateSchema(file)) {
    throw new Error(JSON.stringify(validateSchema.errors!, null, 2))
  }

  return file
}

test('emma config', () => {
  const config = path.join(__dirname, 'fixtures/emma.json')

  expect(parseConfig(config)).toMatchSnapshot()
})
