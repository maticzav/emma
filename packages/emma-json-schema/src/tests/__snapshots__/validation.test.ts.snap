// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emma config 1`] = `
"{
  \\"boilerplates\\": [\\"minimal\\", \\"basic\\", \\"advanced\\"]
}
"
`;
